code/game/objects/items/kitchen.dm

```diff
@@ -102,6 +102,45 @@
 	. = ..()
 	AddElement(/datum/element/easily_fragmented, PLASTIC_BREAK_PROBABILITY)
 
+/obj/item/knife/kitchen/silicon
+	name= "Kitchen Toolset"
+	icon_state = "SiliKnife"	
+	desc = "A breakthrough in synthetic engineering, this tool is a knife programmed to dull when not used for cooking purposes, and can exchange the blade for a rolling pin"
+	force = 0
+	throwforce = 0
+	sharpness = SHARP_EDGED
+	usesound = 'sound/items/drill_use.ogg'
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb_continuous = list("prods", "whiffs", "scratches", "pokes")
+	attack_verb_simple = list("prod", "whiff", "scratch", "poke")
+	tool_behaviour = TOOL_KNIFE
+
+/obj/item/knife/kitchen/silicon/examine()
+	. = ..()
+	. += " It's fitted with a [tool_behaviour == TOOL_KNIFE ? "knife" : "rolling pin"] head."
+
+/obj/item/knife/kitchen/silicon/attack_self(mob/user)
+	playsound(get_turf(user), 'sound/items/change_drill.ogg', 50, TRUE)
+	if(tool_behaviour != TOOL_ROLLINGPIN)
+		tool_behaviour = TOOL_ROLLINGPIN
+		to_chat(user, span_notice("You attach the rolling pin bit to the [src]."))
+		icon_state = "SiliRollingpin"
```

Comment:
especially ditto here with the `snake_case` stuff, this is very cluttered. `sili_rolling_pin` is a lot better.